src/test/java/de/tum/in/www1/artemis/util/CourseTestService.java

```diff
@@ -1524,6 +1526,94 @@ public void searchStudentsInCourse() throws Exception {
         assertThat(tutors).isEmpty();
     }
 
+    public void searchUsersInCourse_searchForAllTutors_shouldReturnAllTutorsAndEditors() throws Exception {
+        var course = database.createCourse();
+        // Test: search for all (no login or name) tutors (tutors includes also editors)
+        var result = searchUsersTest(course, List.of("tutors"), Optional.empty(), numberOfTutors + numberOfEditors, true);
+        assertThat(result.stream().filter(UserPublicInfoDTO::getIsEditor)).hasSize(numberOfEditors);
+        assertThat(result.stream().filter(UserPublicInfoDTO::getIsTeachingAssistant)).hasSize(numberOfTutors);
+    }
+
+    public void searchUsersInCourse_searchForAllInstructor_shouldReturnAllInstructors() throws Exception {
+        var course = database.createCourse();
+        // Test: search for all (no login or name) instructors
+        var result = searchUsersTest(course, List.of("instructors"), Optional.empty(), numberOfInstructors, true);
+        assertThat(result.stream().filter(UserPublicInfoDTO::getIsInstructor)).hasSize(numberOfInstructors);
+    }
+
+    public void searchUsersInCourse_searchForAllStudents_shouldReturnBadRequest() throws Exception {
+        var course = database.createCourse();
+        // Test: Try to search for all students (should fail)
+        searchUsersTest(course, List.of("students"), Optional.empty(), 0, false);
+    }
+
+    public void searchUsersInCourse_searchForStudentsAndTooShortSearchTerm_shouldReturnBadRequest() throws Exception {
+        var course = database.createCourse();
+        // Test: Try to search for all students with a too short search term (at least 3 as students are included) (should fail)
+        searchUsersTest(course, List.of("students"), Optional.of("st"), 0, false);
+    }
+
+    public void searchUsersInCourse_searchForStudents_shouldReturnUsersMatchingSearchTerm() throws Exception {
+        var course = database.createCourse();
+        // Test: Try to search for students with a long enough search term (at least 3 as students are included)
+        // Note: -1 as student1 is the requesting user and will not be returned
+        var result = searchUsersTest(course, List.of("students"), Optional.of(userPrefix + "student"), numberOfStudents - 1, true);
+        assertThat(result.stream().filter(UserPublicInfoDTO::getIsStudent)).hasSize(numberOfStudents - 1);
+    }
+
+    public void searchUsersInCourse_searchForAllTutorsAndInstructors_shouldReturnAllTutorsEditorsAndInstructors() throws Exception {
```

Comment:
This test seems to fail currently.

Reply:
Fixed in a69761e4b537b3c22fe73aa5c44aeb4bdece8035